src/components/tx/TxData.js

```diff
@@ -82,7 +85,25 @@ class TxData extends React.Component {
    * Add all tokens of this transaction (inputs and outputs) to the state
    */
   calculateTokens = () => {
-    this.setState({ tokens: this.props.transaction.tokens });
+    let tokens = this.props.transaction.tokens;
+
+    const metaRequests = tokens.map(token => this.getTokenMetadata(token));
+
+    Promise.all(metaRequests).then((metaResults) => {
+      this.setState({ tokens: metaResults });
+    });
+  }
+
+  /**
+   * Fetch token metadata
+   *
+   * @param {Object} token Token object to be updated
+   */
+  getTokenMetadata = (token) => {
+    return metadataApi.getDag(token.uid).then((data) => {
+      token.meta = data;
```

Comment:
`props` coming from redux are not immutable yet, so we should be careful to not mutate them on the components

I suggest refactoring this to something like:

```
...
return metadataApi.getDag(token.uid).then((data) => ({
  ...token,
  meta: data,
}))
...
```

Reply:
Done.